src/inmanta/server/services/paramservice.py

```diff
@@ -276,14 +280,81 @@ async def list_params(self, env: data.Environment, query: Dict[str, str]) -> Api
         )
 
     @handle(methods_v2.get_facts, env="tid")
-    async def get_facts(self, env: data.Environment, rid: ResourceIdStr) -> List[Parameter]:
+    async def get_facts(self, env: data.Environment, rid: ResourceIdStr) -> List[Fact]:
         params = await data.Parameter.get_list(environment=env.id, resource_id=rid)
-        dtos = [param.to_dto() for param in params]
+        dtos = [param.as_fact() for param in params]
         return dtos
 
     @handle(methods_v2.get_fact, env="tid")
-    async def get_fact(self, env: data.Environment, rid: ResourceIdStr, id: uuid.UUID) -> Parameter:
+    async def get_fact(self, env: data.Environment, rid: ResourceIdStr, id: uuid.UUID) -> Fact:
         param = await data.Parameter.get_one(environment=env.id, resource_id=rid, id=id)
         if not param:
             raise NotFound(f"Fact with id {id} does not exist")
-        return param.to_dto()
+        return param.as_fact()
+
+    @handle(methods_v2.get_parameters, env="tid")
+    async def get_parameters(
+        self,
+        env: data.Environment,
+        limit: Optional[int] = None,
+        first_id: Optional[uuid.UUID] = None,
+        last_id: Optional[uuid.UUID] = None,
+        start: Optional[Union[datetime.datetime, str]] = None,
+        end: Optional[Union[datetime.datetime, str]] = None,
+        filter: Optional[Dict[str, List[str]]] = None,
+        sort: str = "name.asc",
+    ) -> ReturnValue[List[Parameter]]:
+        if limit is None:
+            limit = APILIMIT
+        elif limit > APILIMIT:
+            raise BadRequest(f"limit parameter can not exceed {APILIMIT}, got {limit}.")
+
+        query: Dict[str, Tuple[QueryType, object]] = {}
+        if filter:
+            try:
+                query.update(ParameterFilterValidator().process_filters(filter))
+            except InvalidFilter as e:
+                raise BadRequest(e.message) from e
+        try:
+            parameter_order = ParameterOrder.parse_from_string(sort)
+        except InvalidSort as e:
+            raise BadRequest(e.message) from e
+
+        typed_start, typed_end = None, None
+        if start is not None and start != "None":
+            typed_start = parameter_order.ensure_boundary_type(start)
+        if end is not None and end != "None":
+            typed_end = parameter_order.ensure_boundary_type(end)
```

Comment:
Why do we allow `start` and `end` to be a string as well?

Reply:
I mean. Why do we handle a value `None` similarly as `"None"`

Reply:
The order by column is nullable, and it was possible that the generated url would look like a `start=None&first_id=<some_value>` which is interpreted as the `"None"` string for `start`, but should be understood as `None`.
Based on your comment I updated the part of the code that generates these urls (so e.g. the `start` key is entirely left out when it's value would be `None`), and removed this part.
We can consider handling urls with the `"None"` string to be more robust still, but I'm not sure that's necessary

Reply:
* Interpreting `"None"` in the URL as `None` would cause weird behavior when you filter on the name of a Parameter and the name happens to be `"None"`, right?
* I see you removed the type check completely (`parameter_order.ensure_boundary_type()`) completely. Wouldn't it be better to leave it in?

Reply:
Fair points, agreed :+1: